src/clj/rems/application/model.clj

```diff
@@ -464,10 +466,44 @@
                      :catalogue-item/enabled (:enabled item)
                      :catalogue-item/expired (:expired item)
                      :catalogue-item/archived (:archived item)}
-                    (assoc-some-in [:resource/duo :duo/codes] (seq (enrich-duo-codes duo-codes)))))))
+                    (assoc-some-in [:resource/duo :duo/codes] (seq (duo/enrich-duo-codes duo-codes)))))))
        (sort-by :catalogue-item/id)
        vec))
 
+(defn- enrich-application-duo-matches [application]
+  (if-not (:enable-duo rems.config/env)
+    application
+    (let [application-duos (-> application :application/duo :duo/codes)]
+      (-> application
+          (assoc-in [:application/duo :duo/matches]
+                    (for [resource (:application/resources application)
+                          :let [resource-duos (-> resource :resource/duo :duo/codes)]
+                          :when (seq resource-duos)]
+                      {:resource/id (:resource/id resource)
+                       :resource/duos (for [duo resource-duos]
+                                        (assoc duo :duo/valid? (->> application-duos
+                                                                    (find-first #(= (:id duo) (:id %)))
+                                                                    (duo/check-duo-code duo))))}))))))
```

Comment:
just noticed by checking API response that there is a bit of data duplication with DUO codes: once in application resources under `:resource/duo` and then in duo matches. maybe duo matches could do with `id` and `:duo/valid?`?